### ISSUE TITLE
Unavailable notice for Adsense "Top Earning Pages" widget

### ISSUE THREAD

**Issue by nfmohit** [google/site-kit-wp#6573]
## Feature Description

Since the Adsense `Top Earning Pages` widget works based on a connection between Adsense and UA, and such a connection is not possible at this moment with GA4, it should be replaced with an unavailable notice in the GA4 dashboard view, according to the [Figma designs](https://www.figma.com/file/0UCIE3hPeiu4NYdpAODASd/GA4-Updates?node-id=17%3A1991&t=93hqfio3Z7q3tgxy-0).

Here is a screenshot for visual aid:
![image](https://user-images.githubusercontent.com/20284937/217987005-665cc0f7-dcb6-4145-bda2-8cf53ef8af9c.png)

For this notice, the same component created in #6556 can be re-purposed.

[Here](https://docs.google.com/document/d/14pCNmdPBqAJ-RfPhbHXfaMhGaPK-n-7F-6Sgujd0XCQ/edit#heading=h.hkbagk734m4l) is the relevant section in the design doc.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* When GA4 Reporting is enabled and AdSense is connected and linked to GA, the ["Top Earnings Pages" widget](https://github.com/google/site-kit-wp/blob/f546cb597ae452bbd9b6f0964ceed0dde987011e/assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidget.js#L2) should be replaced with the [notice in the Figma design](https://www.figma.com/file/0UCIE3hPeiu4NYdpAODASd/GA4-Updates?node-id=998%3A8318&t=siCEpmAs40SNspkO-4).
* The notice should contain an "OK, Got it!" button (different from the figma designs, so it's in-line with our existing "OK, Got it!" translation string).
  * When clicked, this button should permanently dismiss the banner. This should cause the widget to return `<WidgetNull />`, meaning the widget area should disappear if no other widgets appear in the area.
* The header in the banner should be: `"Top earning pages are not yet available in Google Analytics 4."`
* The body text in the banner should be: `"Site Kit will notify you as soon as you can connect AdSense and Analytics again."`

## Implementation Brief

* In `assets/js/components/settings/SettingsNotice/SettingsNotice.js`:
    - To avoid duplicating dismissal logic for notices, adapt this component to accept a `dismiss` (string) prop. If a string is passed, use it as the "dismissal key" (see below), and render an "OK, Got it!" button vertically aligned with the whole component.
    - When the dismiss button ("OK, Got it!") is clicked, it should use the `dismissItem` action from the `core/user` store with the `dismissal` prop to persistently store the dismissal in the DB. This functionality can be copied from the `EnableAutoUpdateBannerNotification` component.
* In `assets/js/modules/adsense/constants.js`, create a new `ADSENSE_GA4_TOP_EARNING_PAGES_NOTICE_DISMISSED_ITEM_KEY` that will be used to persistently store the notice dismissal. 
* In `assets/js/modules/adsense/components/dashboard`, create and export a new functional component `DashboardTopEarningPagesWidgetGA4`.
    - This component should accept the `<WidgetNull>` prop.
    - Use the `isItemDismissed` with the above dismissed item key to check if this notice is dismissed. If it is, return `<WidgetNull>`.
    - If the notice is not dismissed, render the `SettingsNotice` component passing the header in the AC as the `notice`. The body text should be passed as the `children` within this component and it will render this below the notice. Some design tweaks might be necessary. Ensure the [existing usage](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/modules-analytics-setup-setupformga4--with-both-tags) of the `<SettingsNoticeMultiRow>` is not affected too much.
    - NOTE: For now, use the above existing component without styling it as per the Figma mock. The styling should be consistently updated for ALL notices in #6702.
* In `assets/js/modules/adsense/index.js`, refactor the `Component` key when registering the `adsenseTopEarningPages` widget. Use the `<GA4DashboardWidgetSwitcher>` to pass the existing `DashboardTopEarningPagesWidget` as the UA component and the new `DashboardTopEarningPagesGA4NoticeWidget` component as the GA4 variant. See #6550 for usage.
* Add a Storybook story and VRT scenario for this GA4 notice variant widget.

### Test Coverage

* No new tests required.

## QA Brief

* Make sure GA4 Reporting is enabled and AdSense is connected and linked to GA
* Check the `Performance over the last 28 days` section - the notice should appear there 
<img width="1560" alt="Screenshot 2023-04-06 at 17 15 17" src="https://user-images.githubusercontent.com/16494321/230437470-a655d8da-896b-4bb3-a1a8-d200fcfcbfd3.png">

## Changelog entry

* Add the new GA4 version of the Top Earning Pages widget.

**Comment by eugene-manuilov** [comment]
@jimmymadon 

> * To avoid duplicating dismissal logic for notices, adapt this component to accept a `dismiss` (boolean) prop. If true, it should render a "OK, Got it!" button vertically aligned with the whole component.
> * This component should also accept a `dismissalKey` (string) prop. When the dismiss button ("OK, Got it!") is clicked, it should use the `dismissItem` action from the `core/user` store with the `dismissalKey` to persistently store the dismissal in the DB. This functionality can be copied from the `EnableAutoUpdateBannerNotification` component.

We can use just one prop `dismiss` but make it be a string. If it's empty, no need to render the dismissal button. If not empty, then render the button and use it as the dismissal item slug.

> In `assets/js/components/settings/SettingsNotice/SettingsNotice.js`:
> * To avoid duplicating dismissal logic for notices, adapt this component to accept a `dismiss` (boolean) prop. If true, it should render a "OK, Got it!" button vertically aligned with the whole component.

If we adapt it to be dismissable and going to use it in other places, then that component should be renamed and moved from the `SettingsNotice` group to something not settings specific. In the `components` folder we have the legacy `Alert` component that we have forgotten to delete and that is not used anywhere. Let's rename the `SettingsNotice` component to be `Alert` and move it to the `components` folder. The legacy content can be removed.

> In `DashboardTopEarningPagesWidget.js`:
> * If the `ga4Reporting` feature flag is enabled, use the `isGA4DashboardView` from the `modules/analytics` datastore to see if the dashboard is displaying GA4 data. If it is, return a new component, say `GA4TopEarningPagesNoticeWidget` passing the `<WidgetNull>` prop to it.

I think instead of creating a new component for notice displaying purposes only, we need to create the GA4 copy of the widget and register the same way how we register other GA4 widgets to keep our approach consistent across all widgets. See #6550 for more details. The existing `DashboardTopEarningPagesWidget` widget should remain the same.

**Comment by eugene-manuilov** [comment]
@jimmymadon 

> > Let's rename the SettingsNotice component to be Alert and move it to the components folder.
> 
> I would suggest renaming this component to `Notice` as "Alert" sounds a bit more like a JS Alert popup. I've added a note to #6702 as well to include the refactoring there.

My main concern here is that it won't be clear which notice it is because we already have quite a lot different notices and having one more that just says "Notice" might be confusing. Maybe let's call it `AlertNotice` then? 😄 

![image](https://user-images.githubusercontent.com/873049/226597358-09e073d5-d0bf-45b9-b30b-aadd85fc6109.png)

> > then that component should be renamed and moved from the SettingsNotice group to something not settings specific.
> 
> Since this issue had some refactoring already, I was skeptical of increasing the scope of this issue and test all existing `SettingsNotice` usage. I thought we could do the move refactor in #6702 as we will be testing all notices there for style changes throughout the plugin.

Sound good to me, thanks 👍 

> * In `assets/js/modules/adsense/components/dashboard`, create and export a new functional component `DashboardTopEarningPagesGA4NoticeWidget`.

The widget should be named exactly as its UA counterpart but with GA4 suffix, the same as other widgets: `DashboardTopEarningPagesWidgetGA4`

**Comment by eugene-manuilov** [comment]
Thanks, @jimmymadon. IB ✔️

**Comment by eugene-manuilov** [comment]
@sashadoes, I approved and merged your PR, but then realised that one important piece had been missed in IB and in your implementation - the GA4 version of the widget should preserve the same layout but instead of showing the table with results it should show the notice. This is a bit tricky thing because we will probably need to update the `ReportTable` component to display an alternative state when there is no rows to let us use the notice there.

Just to repeat, we need the new GA4 version to return the same as in its UA version but using the `ReportTable` component with empty rows and the notice to be displayed by the table when there are no rows. Could you please create a new PR for it? https://github.com/google/site-kit-wp/blob/d92dc83492d230c4da1515692818bfa03160395c/assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidget.js#L207-L219

**Comment by sashadoes** [comment]
Hi @eugene-manuilov, thanks for the feedback.
Sorry, I am not sure what is needed here. 
Could you please describe some steps in IB please? 
Thanks in advance.

**Comment by aaemnnosttv** [comment]
Thanks @eugene-manuilov – I just spoke to @sashadoes about this – I think the important part here is the addition of the wrapping `<Widget>`, whether or not it uses an enhanced `<ReportTable>` here is a bit secondary – if it's possible sure but we probably don't need to invest too much effort to match that aspect of the design exactly.

**Comment by eugene-manuilov** [comment]
@aaemnnosttv that's sounds good to me, I asked for the report table changes because i noticed that we have column headers on the screenshot in the ticket description. I assume that QA engineers will ask questions why those are missing in the actual implementation.

**Comment by aaemnnosttv** [comment]
It might still be possible to use the `<ReportTable>` but I don't think it's a critical detail for a placeholder component like that. It's worth looking into, but mostly something we don't want this issue to get held up by.

**Comment by eugene-manuilov** [comment]
@sashadoes I see you set QAB as not applicable for this ticket. Why is that? Could you please update QAB to have the correct steps to QA your changes? Thanks.

**Comment by mohitwp** [comment]
### QA Update ❌ 

@sashadoes 

Tested on main environment.

**Issues -** 

- Notice font size and font weight is not same as figma design.
- Icon is different from Figma design.
- Font color is different.
- Button is not aligned.
- Widget have no title.

<details>

**Notice -**

![image](https://user-images.githubusercontent.com/94359491/230622499-4a1910f5-b852-4137-af0a-568cc8b53bac.png)

**Figma -**

![image](https://user-images.githubusercontent.com/94359491/230622576-e97f4617-ae09-40c2-a87c-f9aa5124c585.png)

</details>

**Comment by sashadoes** [comment]
Hi @mohitwp, thanks for the review and comment. 

My understanding is that the design will be implemented in #6702, at least the IB says so. 
Was it a mistake and should I also implement the design here? cc @aaemnnosttv

**Comment by mohitwp** [comment]
### QA Update ✅ 

Thanks @sashadoes .I'm putting a note on #6702 so that we make sure all design related issues get address in related PR. 

- Tested on main.
- Unavailable notice for AdSense is showing under "Top Earning Widget" as per AC and Figma design.
- Note : There are design related issues as mentioned [above](https://github.com/google/site-kit-wp/issues/6573#issuecomment-1500325105) . Those issues will be address by #6702.


![image](https://user-images.githubusercontent.com/94359491/230622499-4a1910f5-b852-4137-af0a-568cc8b53bac.png)
